### PR TITLE
Fix:Addon OptionsError is neither logged, nor does it stop mitmproxy

### PR DESCRIPTION
**Pull request by sarthak212**
#3558

### CODE REVIEW

**Comment by sarthak212** [comment]
@mhils I think Configure in scripts is not raising any type of error.

**Comment by mhils** [comment]
Thanks! Can we start by adding a test that demonstrates the wrong behavior?

**Comment by sarthak212** [comment]
Yeah but should i write custom dummy script to pass as argument or i use prebuild script?

File: test/mitmproxy/data/addonscripts/configure.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+        )
+
+    def configure(self, updates):
+        raise exceptions.OptionsError("Options Error")
+
+    def response(self, flow):
+        if ctx.options.testheader is not None:
+            flow.response.headers["testheader"] = str(ctx.options.testheader)
+
+
```
Comment (mhils) [inline test/mitmproxy/data/addonscripts/configure.py]: This really isn't needed here, let's remove it.

File: test/mitmproxy/data/addonscripts/configure.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import typing
+
+from mitmproxy import ctx
+from mitmproxy import exceptions
+
+
+class TestHeader:
```
Comment (mhils) [inline test/mitmproxy/data/addonscripts/configure.py]: Give this thing a better name. :)

**Comment by sarthak212** [comment]
@mhils I have done changes.

File: test/mitmproxy/data/addonscripts/configure.py (hunk 2 above)
Comment (sarthak212) [reply test/mitmproxy/data/addonscripts/configure.py]: I changed the name to "OptionAddon" if that doesn't fit can you give me a little hint, I am not too good with names.

**Comment by mhils** [comment]
Great, thanks! 🍰

**Pull request merged**